src/index.ts

```diff
@@ -1412,7 +1414,29 @@ export abstract class PinejsClientCore<PinejsClient> {
 
 		// Do not await this._request result, so that we can preserve
 		// the potentially enhanced promise-like result.
-		return this.callWithRetry(() => this._request(opts), retry);
+		return this.callWithRetry(async () => {
+			if (body) {
+				const blobKeys = Object.keys(body).filter(
+					(key) => body[key] instanceof Blob,
+				);
+				if (blobKeys.length > 0) {
+					opts.files = blobKeys.map(async (key) => {
```

Comment:
Is this intended to match the interface of a specific request library?

Reply:
Yes. Matching this to the structure of `req.files` which is the following:
```
[
  {
    "fieldname": "logo",
    "originalname": "logo.png",
    "encoding": "7bit",
    "mimetype": "image/png",
    "buffer": {
      "type": "Buffer",
      "data": [] // being the Blob buffer
    },
    "size": 0
  }
]
```